File: CRUD/C#/Aplicacion con SQL Server V2/Aplicacion/Forms/FormHome.cs
// Written By Ismael Heredia in the year 2017

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Aplicacion.Data;

namespace aplicacion
{
    public partial class FormHome : Form
    {
        UsuarioDatos usuarioDatos = new UsuarioDatos();
        String usuario = "";

        public FormHome(string usuario_recibido)
        {
            InitializeComponent();
            usuario = usuario_recibido;
        }

        private void FormHome_Load(object sender, EventArgs e)
        {
            if (usuarioDatos.get_user_type(usuario)=="Administrador")
            {
                ItemUsuarios.Enabled = true;
            }
            else
            {
                ItemUsuarios.Enabled = false;
            }
        }

        private void ItemProductos_Click(object sender, EventArgs e)
        {
            FormProductos form = new FormProductos();
            form.Show();
        }

        private void ItemProveedores_Click(object sender, EventArgs e)
        {
            FormProveedores form = new FormProveedores();
            form.Show();
        }

        private void ItemSalir_Click(object sender, EventArgs e)
        {
            if (MessageBox.Show("¿ Esta seguro de salir del programa ?", "Salir", MessageBoxButtons.YesNo, MessageBoxIcon.Stop, MessageBoxDefaultButton.Button2) == DialogResult.Yes)
            {
                Application.Exit();
            }
        }

        private void FormHome_FormClosing(object sender, FormClosingEventArgs e)
        {
            Application.Exit();
        }

        private void ItemCambiarNombreDeUsuario_Click(object sender, EventArgs e)
        {
            FormCambiarUsuario formCambiarUsuario = new FormCambiarUsuario(usuario);
            formCambiarUsuario.Show();
        }

        private void ItemCambiarClave_Click(object sender, EventArgs e)
        {
            FormCambiarClave formCambiarContraseña = new FormCambiarClave(usuario);
            formCambiarContraseña.Show();
        }

        private void ItemUsuarios_Click(object sender, EventArgs e)
        {
            FormUsuarios formUsuarios = new FormUsuarios();
            formUsuarios.Show();
        }

        private void tsEstadisticas_Click(object sender, EventArgs e)
        {
            FormReporte formReporte = new FormReporte();
            formReporte.Show();
        }
    }
}

File: CRUD/VB.Net/Aplicacion con SQL Server V2/Sistema/Sistema/Forms/FormHome.vb
' Written By Ismael Heredia in the year 2017

Imports Sistema.Aplicacion.Data

Public Class FormHome

    Dim usuarioDatos As New UsuarioDatos
    Dim usuario As String = ""

    Public Sub New(ByVal usuario_recibido As String)
        InitializeComponent()
        usuario = usuario_recibido
    End Sub

    Private Sub frmHome_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If usuarioDatos.get_user_type(usuario) = "Administrador" Then
            ItemUsuarios.Enabled = True
        Else
            ItemUsuarios.Enabled = False
        End If
    End Sub

    Private Sub ItemProductos_Click(sender As Object, e As EventArgs) Handles ItemProductos.Click
        Dim form As New FormProductos()
        form.Show()
    End Sub

    Private Sub ItemProveedores_Click(sender As Object, e As EventArgs) Handles ItemProveedores.Click
        Dim form As New FormProveedores()
        form.Show()
    End Sub

    Private Sub ItemUsuarios_Click(sender As Object, e As EventArgs) Handles ItemUsuarios.Click
        Dim formUsuarios As New FormUsuarios()
        formUsuarios.Show()
    End Sub

    Private Sub ItemCambiarUsuario_Click(sender As Object, e As EventArgs) Handles ItemCambiarUsuario.Click
        Dim formCambiarUsuario As New FormCambiarUsuario(usuario)
        formCambiarUsuario.Show()
    End Sub

    Private Sub ItemCambiarClave_Click(sender As Object, e As EventArgs) Handles ItemCambiarClave.Click
        Dim formCambiarContraseña As New FormCambiarClave(usuario)
        formCambiarContraseña.Show()
    End Sub

    Private Sub ItemSalir_Click(sender As Object, e As EventArgs) Handles ItemSalir.Click
        If MessageBox.Show("¿ Esta seguro de salir del programa ?", "Salir", MessageBoxButtons.YesNo, MessageBoxIcon.Stop, MessageBoxDefaultButton.Button2) = DialogResult.Yes Then
            Application.Exit()
        End If
    End Sub

    Private Sub ItemEstadisticas_Click(sender As Object, e As EventArgs) Handles ItemEstadisticas.Click
        Dim formReporte As New FormReporte()
        formReporte.Show()
    End Sub

    Private Sub frmHome_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        Application.Exit()
    End Sub

End Class